lib/ip_info/parser.rb

```diff
@@ -5,17 +5,25 @@ class API
     module Parser
       class InvalidParamsError < ArgumentError; end
 
-      def parse_response(response)        
-        result = convert_keys response
-        raise InvalidParamsError.new(result[:status_message]) if result[:status_code] == "ERROR"
-        result
+      def parse_response(response)
+        convert_keys(response).tap do |result|
+          raise InvalidParamsError.new(result[:status_message]) if result[:status_code] == "ERROR"
+        end
       end
 
       private
-        def convert_keys(response)
-          raise InvalidParamsError.new("Params must be presence") if response.nil?
-          response.inject({}) {|hsh, (key,value)| hsh[key.to_s.gsub(/[A-Z]/){|s| '_' + s.downcase}.gsub(/^_/, '').to_sym] = value; hsh}
+
+      def convert_keys(response)
+        raise InvalidParamsError.new("Params must be present") if response.nil?
+
+        response.each_with_object({}) do |(key, value), hsh|
+          hsh[to_snake_case(key)] = value
         end
+      end
+
+      def to_snake_case(key)
+        key.to_s.gsub(/[A-Z]/) { |s| '_' + s.downcase }.gsub(/^_/, '').to_sym
```

Comment:
I have just moved this code away from `convert_keys` method. Apart from making it more readable I think it is not handling correctly acronyms (usually they should not be split).

When it comes to your solution, I don't think `Enumerator` has `join` method. Maybe you meant `split` instead of `gsub` (it would still not work correctly)?